projectdoc/analyse.tex

```diff
@@ -4,15 +4,137 @@ \section{Analyse}
+
+Dagegen ist die Vue.js-Komponente nur ein einfacher Wrapper um die API von \emph{Mapbox GL}.
+So kann z.B. ein zusätzlicher Layer nicht direkt im Template eingebunden werden, sondern muss über Event-Handling nach dem Laden der Karte hinzugefügt werden.
+Dies macht das dynamische Hinzufügen und Entfernen von Karten-Elementen, was wir in unserer Applikation viel benötigen, sehr mühsam.
+
+\subsubsection{Lernkurve}
+\label{Analyse Framework:Lernkurve}
+
+Die Lernkurve beschreibt under anderem, wie viel Zeit benötigt wird, sich in die neue Technologie einzuarbeiten.
+Dies ist durchaus ein subjektives Kriterium und hängt von unserem bestehenden Erfahrung mit anderen Web-Frameworks ab.
+
+Die beiden sehr ähnlichen Philosophien von React und Vue.js weichen in gewissen Punkten stark von uns vertrauten Praktiken ab, so etwa die inhärente Kopplung von Darstellung und Logik, die in anderen Modellen wie MVC (Model-View-Controller) aufgespalten wird.
+Insofern benötigen beide Varianten eine gewisse Einarbeitungszeit, um sich mit diesen Konzepten vertraut zu machen.
+
+Bei React kommt zusätzlich die neue Syntax JSX~\cite{jsx} dazu, die zwar etwas angenehmer zu lesen ist, aber in vielen kleinen Punkten von bereits Bekanntem abweicht.
+So mussten z.B. HTML-Attribute wie \code{for} umbenannt werden, um nicht eine Namenskollision mit JavaScript zu verursachen.
+
+Die Template-Syntax von Vue.js ist deutlich vertrauter, auch weil sie sehr dem Stil von anderen Template-Engines ähnelt.
+Vue.js erlaubt auch einiges an ''Syntactic Sugar'' in den Templates.
+So ist simples Data-Binding mit Event-Handling direkt im Template möglich, während bei React dies separat behandelt werden muss.
+
+Aus dieser Einschätzung kann man sagen, dass die Lernkurve bei Vue.js etwas flacher als bei React ist.
+
+\subsubsection{Tooling}
+\label{Analyse Framework:Tooling}
+
+Da sowohl React wie auch Vue.js auf NodeJS aufsetzen und den Paketmanager NPM benutzen, ist das Tooling sehr ähnlich.
+
+Um eine neue Applikation zu erstellen, bietet React das Tool \emph{create-react-app}~\cite{create_react_app} an.
+Beim Ausführen kann man angeben, welche zusätzlichen Libraries (z.B. für Unit-Testing) man verwenden möchte.
+Während der Entwicklung kann ausserdem ein lokaler Server gestartet werden, der jeweils automatisch das Browser-Fenster neu ladet, wenn sich der Source-Code geändert hat.
+
+Vue.js bietet mit \emph{vue-cli}~\cite{vue_cli} ein ähnliches Tool an.
+Um eine neue Applikation aufzusetzen, kann zwischen verschiedenen Templates gewählt werden.
+Es gibt verschiedene Templates für z.B. unterschiedliche Grössen der geplanten Applikation.
+Während der Entwicklung bietet \emph{vue-cli} die praktisch identischen Möglichkeiten wie das Tool für React.
+
+Insgesamt kann zwischen den beiden Varianten in Sachen Tooling praktisch keinen Unterschied festgestellt werden.
+
+
+\subsubsection{Auswertung}
+\label{Analyse Framework:Auswertung}
+
```

Comment:
Wollen wir nicht noch ein Fazit ziehen, auf was jetzt die Entscheidung gefallen (natürlich ist es in der Tabelle ersichtlich)

Reply:
done